core/runtime/binaryen/runtime_environment.hpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_RUNTIME_BINARYEN_RUNTIME_ENVIRONMENT
+#define KAGOME_CORE_RUNTIME_BINARYEN_RUNTIME_ENVIRONMENT
+
+#include <memory>
+
+#include <boost/optional.hpp>
+
+#include "outcome/outcome.hpp"
+
+namespace kagome::common {
+  class Buffer;
+}
+
+namespace kagome::storage::trie {
+  class TopperTrieBatch;
+  class TrieBatch;
+}  // namespace kagome::storage::trie
+
+namespace kagome::runtime {
+  class WasmMemory;
+}
+
+namespace kagome::runtime::binaryen {
+  class RuntimeExternalInterface;
+  class WasmModuleInstance;
+  class WasmModule;
+
+  /**
+   * Runtime environment is a structure that contains data necessary to operate
+   * with Runtime: memory, WASM module, storage batch
+   */
+  class RuntimeEnvironment {
+   public:
+    static outcome::result<RuntimeEnvironment> create(
+        const std::shared_ptr<RuntimeExternalInterface> &rei,
+        const std::shared_ptr<WasmModule> &module,
+        const common::Buffer &state_code);
+
+    RuntimeEnvironment(RuntimeEnvironment &&) = default;
+    RuntimeEnvironment &operator=(RuntimeEnvironment &&) = default;
+
+    RuntimeEnvironment(const RuntimeEnvironment &) = delete;
```

Comment:
why do we need to delete them explicitly?

Reply:
Rule of five, just to make things more clear.

Reply:
@masterjedy look at [The rule of three/five/zero](https://en.cppreference.com/w/cpp/language/rule_of_three)

Reply:
Russian article on [Habr](https://ru.wikipedia.org/wiki/%D0%9F%D1%80%D0%B0%D0%B2%D0%B8%D0%BB%D0%BE_%D1%82%D1%80%D1%91%D1%85_(C%2B%2B))